source/appModules/powerpnt.py

```diff
@@ -1043,44 +1046,123 @@ def script_enterChart(self, gesture):
 
 
 class TextFrameTextInfo(textInfos.offsets.OffsetsTextInfo):
-	def _getCaretOffset(self):
+	def _getCaretOffset(self) -> int:
 		return self.obj.documentWindow.ppSelection.textRange.start - 1
 
-	def _getSelectionOffsets(self):
+	def _getSelectionOffsets(self) -> tuple[int, int]:
 		sel = self.obj.documentWindow.ppSelection.textRange
 		start = sel.start - 1
 		end = start + sel.length
 		return start, end
 
-	def _getTextRange(self, start, end):
-		# #4619: First let's "normalise" the text, i.e. get rid of the CR/LF mess
-		text = self.obj.ppObject.textRange.text
-		text = text.replace("\r\n", "\n")
-		# Now string slicing will be okay
-		text = text[start:end].replace("\x0b", "\n")
-		text = text.replace("\r", "\n")
-		return text
+	def _getPptTextRange(
+		self,
+		start: int,
+		end: int,
+		clamp: bool = False,
+	) -> comtypes.client.lazybind.Dispatch:
+		"""
+		Retrieves a text range from the PowerPoint object, with optional clamping of the start and end indices.
+		:param start: The starting character index of the text range (zero based).
+		:param end: The ending character index of the text range(zero based).
+		:param clamp: If True, the start and end indices will be clamped to valid values within the text range. Defaults to False.
+		:returns: The text range object as a comtypes.client.lazybind.Dispatch object.
+		:raises ValueError: If the start index is greater than the end index.
+		:note: For more information about text ranges, see https://learn.microsoft.com/en-us/office/vba/api/powerpoint.textrange
```

Comment:
I wonder if it's worth linking this in the class docstring instead, since it's used widely throughout the class, not just in this method. What do you think?